### PR TITLE
Qualcomm AI Engine Direct - Requantization Mechanism Implementation

### PR DESCRIPTION
**Pull request by winskuo-quic**
Summary:

- Implement requantization so mix quantization ( e.g., 8bit quant + 16 bit quant) can be properly delegated to QNN.
- Reusing test_qnn_backend_view_permute_matmul unit test to check mix quantization is working as expected.
- Added etdump logic back to qnn_executor_runner that was deleted unintentionally during this PR: https://github.com/pytorch/executorch/commit/a531ca5a5ab5b5e503739a40a7cbec2e46767f0b#diff-f3647de74042ac9a417e2d4000a6f2db00c22c89fd028e9433d3c79ffb7d56f6
- Refactor common arguments in VIT.

### CODE REVIEW

**Comment by winskuo-quic** [comment]
Hi @cccclai,
This PR is used to support mix quantization and fixing minor issues that's causing UT failures.
Please have a look on this PR.
Thanks

**Comment by cccclai** [comment]
Hey would you mind explaining a bit what does requantization mean and how it's different than mix precision quantization? Does it mean we'll do 8bit quantization first then 16bit (maybe skip the 8bit quantized ops)?

**Comment by winskuo-quic** [comment]
> Hey would you mind explaining a bit what does requantization mean and how it's different than mix precision quantization? Does it mean we'll do 8bit quantization first then 16bit (maybe skip the 8bit quantized ops)?

Hi Chen,
Thanks for the response.
Yes.
There are cases where some ops have output of 8 bit, however, the input only accepts 16bit. In this case, we will requantize the model so that the input can properly accept a 16bit input from the op user. This scenario where we have different quant config is mix quantization.

**Comment by cccclai** [comment]
> > Hey would you mind explaining a bit what does requantization mean and how it's different than mix precision quantization? Does it mean we'll do 8bit quantization first then 16bit (maybe skip the 8bit quantized ops)?
> 
> Hi Chen, Thanks for the response. Yes. Requantization should be the same as mix precision quantization. There are cases where some ops have output of 8 bit, however, the input only accepts 16bit. In this case, we will requantize the model so that the input can properly accept a 16bit input from the op user. This scenario where we have different quant config is mix quantization.

Thank you for the explanation! Is there an example somewhere to show how to do it, maybe a test case?

File: backends/qualcomm/tests/test_qnn_delegate.py [hunk 1]
```diff
@@ -981,7 +981,7 @@ def test_qnn_backend_view_permute_matmul(self):
         sample_input = (torch.randn([1, 8, 512]), torch.randn([1, 2, 8, 256]))
         module = self.get_qdq_module(module, sample_input)
         self.lower_module_and_test_output(module, sample_input)
-        # check if requantization work
+        # check if requantization work by reusing the 8bit qdq module
```
Comment (winskuo-quic) [inline backends/qualcomm/tests/test_qnn_delegate.py:984]: @cccclai 
Yes. Please refer to the unit test test_qnn_backend_view_permute_matmul. It demonstrates how mix quantization works.

**Comment by winskuo-quic** [comment]
> > > Hey would you mind explaining a bit what does requantization mean and how it's different than mix precision quantization? Does it mean we'll do 8bit quantization first then 16bit (maybe skip the 8bit quantized ops)?
> > 
> > 
> > Hi Chen, Thanks for the response. Yes. Requantization should be the same as mix precision quantization. There are cases where some ops have output of 8 bit, however, the input only accepts 16bit. In this case, we will requantize the model so that the input can properly accept a 16bit input from the op user. This scenario where we have different quant config is mix quantization.
> 
> Thank you for the explanation! Is there an example somewhere to show how to do it, maybe a test case?

Yes.
Please refer to the unit test  TestQNNQuantizedModel.test_qnn_backend_view_permute_matmul under test_qnn_delegate.py.
https://github.com/pytorch/executorch/pull/2823/files#diff-3a76d2f6f72394bf64270a625d31c16560eb7d3b855297352aae423a27f6f59fR979
It demonstrates how mix quantization works.

**Review by kirklandsign — APPROVED** [review]

**Comment by cccclai** [comment]
Hey do you mind rebasing the change? The CI was broken on the base commit and should be fixed now.

**Comment by winskuo-quic** [comment]
> Hey do you mind rebasing the change? The CI was broken on the base commit and should be fixed now.

Thanks for the friendly reminder. I have just rebased.

**Pull request closed without merging**